cgroup1/pids.go

```diff
@@ -67,16 +66,10 @@ func (p *pidsController) Stat(path string, stats *v1.Metrics) error {
 	if err != nil {
 		return err
 	}
-	var max uint64
-	maxData, err := os.ReadFile(filepath.Join(p.Path(path), "pids.max"))
+	max, err := readUint(filepath.Join(p.Path(path), "pids.max"))
```

Comment:
Isn't the behavior different here? `readUint` should fail if "max" exists in the file as it only knows how to parse integers, so we'd bail on line 71. Whereas before we'd carry on to filling in the stats, granted the `max` variable will be left as 0 it seems if "max" was in pids.max, but point being this method would still succeed

Reply:
Right; I changed readUint to return MaxUint64 now. The behavior is still different but it is consistent with cgroups2. I think this change should be fine now.

Reply:
Yea it is odd to me how we returned 0 here before.. would need to see if any users relied on this behavior with a github search maybe..

Reply:
(I think returning MaxUint64 makes sense)

Reply:
I think it makes sense at least for v3; v1 is imported by so many packages that I am not confident.

1. https://pkg.go.dev/github.com/containerd/cgroups?tab=importedby
2. https://pkg.go.dev/github.com/containerd/cgroups/v3@v3.0.1?tab=importedby


Thinking of reverting back to 0 and adding a comment to say that we preserve it for backward compatibility.
What do you think?

Reply:
I'd err on the side of staying safe, so your reasoning seems sane and I agree (also sorry for the radio silence here 🫠)

Reply:
I suppose 0 could be interpreted as no limit, so that might've been the rationale

Reply:
Zero wouldn't be a valid value anyway as zero, so that's a fair rationale.

Reply:
@dcantah No worries! thanks for the review! this was a really fun exercise :)
thanks @mmerkes for the review and helping building understanding